### PR TITLE
Add example, setup instructions, and explanation to README

### PR DESCRIPTION
**Pull request by boatbomber**
Read the rendered version instead of the raw markdown:
https://github.com/Ho1yShif/dinner/blob/update-documentation/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
...
+1. Create a Google Sheet with two sheets, `Meals` and `Shopping`
+
+   ![image](https://github.com/Ho1yShif/dinner/assets/40185666/2c688c51-7fac-4ad9-b3f6-7f885b261be1)
+
+2. Set up a Google Cloud service account with access to the Google Sheets API and create a new JSON key ([Learn more about Service Accounts.](https://cloud.google.com/iam/docs/service-account-overview?authuser=4))
+   - Replace the `\n` in the key with `\\n` so Python's json library can parse it
+   - Encode the santized key JSON in base64 so it can be easily stored in GitHub Secrets
+
+3. Fork this repo
+   - You may want to edit the `dinners.json` file to include your own preferred meals
```
Comment (Ho1yShif) [inline README.md]: Customize your meal plan using the `dinners.json` file

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+
+4. Set repo actions Secrets
+    - `SERVICE_ACCOUNT` is the Google service account JSON file encoded in base64
+    - `SPREADSHEET_ID` is the ID of your Google Sheet (the long string in the URL)
+
+## How it works
+
+A GitHub Action is scheduled each week and runs the `plan_dinners.py` script. This script will read the `dinners.json` file and randomly select meals for each day of the week. It won't pick the same category of meal more than once to avoid making similar meals every day.
```
Comment (Ho1yShif) [inline README.md]: A GitHub Action is scheduled to run the `plan_dinners.py` script weekly at 3 AM on Saturday

Comment (Ho1yShif) [inline README.md]: Each meal category will be represented only once per week to avoid repetition

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+
+It will then update the `Meals` sheet with the chosen meals and update the `Shopping` sheet with the ingredients needed for those meals. The shopping list will also include a reminder to check if you are stocked on staples like eggs, bread, oil, etc.
```
Comment (Ho1yShif) [inline README.md]: The GitHub action will then...

**Review by Ho1yShif — COMMENTED** [review]
Beautiful work; I made a few tiny comments :)

File: README.md (hunk 3 above)
Comment (boatbomber) [reply README.md]: Actually the It refers to the script like the previous paragraph. I agree it's unclear though.

**Review by Ho1yShif — APPROVED** [review]

**Pull request merged**